openpype/hosts/maya/plugins/publish/validate_shader_name.py

```diff
@@ -60,19 +61,18 @@ def get_invalid(cls, instance):
             )
 
             for shader in shaders:
-                m = r.match(cls.regex, shader)
+                m = re.match(cls.regex, shader)
                 if m is None:
                     invalid.append(shape)
-                    cls.log.error(
-                        "object {0} has invalid shader name {1}".format(shape,
-                                                                        shader)
-                    )
+                    cls.log.error(error_message.format(shape, shader))
                 else:
-                    if 'asset' in r.groupindex:
+                    if 'asset' in regex_compile.groupindex:
                         if m.group('asset') != asset_name:
                             invalid.append(shape)
-                            cls.log.error(("object {0} has invalid "
-                                           "shader name {1}").format(shape,
-                                                                     shader))
+                            message = error_message
+                            message += " with missing asset name \"{2}\""
+                            cls.log.error(
+                                message.format(shape, shader, asset_name)
+                            )
```

Comment:
Technically the regex could have been altered to not enforce the asset name to be a part of it?